tests/FactoryTest.php

```diff
@@ -147,17 +152,17 @@ public function testAddDataSource()
         ]);
         $factory = new DataSourceFactory($driveFactoryManager);
 
-        $driver = $this->createMock('FSi\Component\DataSource\Driver\DriverInterface');
-        $datasource = $this->createMock('FSi\Component\DataSource\DataSource', [], [$driver]);
-        $datasource2 = $this->createMock('FSi\Component\DataSource\DataSource', [], [$driver]);
+        $driver = $this->createMock(DriverInterface::class);
+        $datasource = $this->createMock(DataSource::class, [], [$driver]);
```

Comment:
`createMock` accepts, only one parameter, you will need to add constructor arguments here through the `getMockBuilder`